src/backend/bs_repserver

```diff
@@ -3154,24 +3154,33 @@ sub getjobhistory {
   if ($cgi->{'code'} && @{$cgi->{'code'}} == 1 && $cgi->{'code'}->[0] eq 'lastfailures') {
     return getlastfailures($cgi, $projid, $repoid, $arch);
   }
+  my $idx = 0;
+  my %indices = map {$_ => $idx++} @$BSXML::jobhistlay;
+  my $packidx = %indices{'package'};
+  my $codeidx = %indices{'code'};
+  my $endtimeidx = %indices{'endtime'};
   if ($cgi->{'package'} && $cgi->{'code'}) {
     my %packid = map {$_ => 1} @{$cgi->{'package'}};
     my %code = map {$_ => 1} @{$cgi->{'code'}};
-    $filter = sub {$packid{$_[0]->{'package'}} && $code{$_[0]->{'code'}}};
+    $filter = sub {$packid{$_[$packidx]} && $code{$_[$codeidx]}};
```

Comment:
But the filter is only called with one argument, so what is `$_[...]` supposed to do?

And what about the decoding? If you want to skip the decoding you need to encode the values in the filter.